lib/core/jtrace.c

```diff
@@ -550,6 +742,275 @@ j_trace_enter(gchar const* name, gchar const* format, ...)
 		G_UNLOCK(j_trace_echo);
 	}
 
+	if (j_trace_flags & J_TRACE_ACCESS)
+	{
+		if (strcmp(name, "ping") == 0)
+		{
+			va_list args;
+			va_start(args, format);
+			g_free(trace_thread->client.program_name);
+			trace_thread->client.program_name = strdup(va_arg(args, const char*));
+			trace_thread->client.uid = va_arg(args, guint32);
+			va_end(args);
+		}
+		else if (strncmp(name, "backend_", 8) == 0)
+		{
+			int type = 0;
+			if (trace_thread->access.db.type == NULL) /// \TODO more precise checking?
+			{
+				JConfiguration* config = j_configuration_new();
+
+				trace_thread->access.db.type = strdup(j_configuration_get_backend(config, J_BACKEND_TYPE_DB));
+				trace_thread->access.db.config_path = strdup(j_configuration_get_backend_path(config, J_BACKEND_TYPE_DB));
+
+				trace_thread->access.kv.type = strdup(j_configuration_get_backend(config, J_BACKEND_TYPE_KV));
+				trace_thread->access.kv.config_path = strdup(j_configuration_get_backend_path(config, J_BACKEND_TYPE_KV));
+
+				trace_thread->access.object.type = strdup(j_configuration_get_backend(config, J_BACKEND_TYPE_OBJECT));
+				trace_thread->access.object.config_path = strdup(j_configuration_get_backend_path(config, J_BACKEND_TYPE_OBJECT));
+
+				j_configuration_unref(config);
+			}
+			name += 8;
```

Comment:
Hard to catch should the prefix change later on.
Maybe add defines for the prefix and prefix length? This would still avoid pattern matching overhead.

Reply:
also used later on: is it feasible to split the string (even before the match against 'backend_') on '_' and compare parts instead?

Reply:
We cannot split at every `_` because sometimes it appears in the name, like `get_all`. And doing it before the check for `backend_` seems like a lot of potential string comparing and allocating because the name is constant. Therefore, we cannot simply add `\0`